python/cucim/src/cucim/core/operations/morphology/_pba_2d.py

```diff
@@ -119,27 +151,34 @@ def _pba_2d(arr, sampling=None, return_distances=True, return_indices=False,
         # if len(sampling) != 2:
         #     raise ValueError("sampling must be a sequence of two values.")
 
-    padded_size = math.ceil(max(arr.shape) / block_size) * block_size
     if block_params is None:
-        # should be <= size / 64. sy must be a multiple of m1
-        m1 = max(1, min(padded_size // block_size, 32))
+        padded_size = math.ceil(max(arr.shape) / block_size) * block_size
 
+        # should be <= size / 64. sy must be a multiple of m1
+        m1 = padded_size // block_size
         # size must be a multiple of m2
-        m2 = max(1, min(padded_size // block_size, 32))
+        m2 = max(1, min(padded_size // block_size, block_size))
         # m2 must also be a power of two
         m2 = 2**math.floor(math.log2(m2))
         if padded_size % m2 != 0:
             raise RuntimeError("error in setting default m2")
-
-        # should be <= 64. texture size must be a multiple of m3
+        # should be <= 64. image size must be a multiple of m3
```

Comment:
Do we need  a statement for checking the value is `<=64` somewhere?

Reply:
I think that was an outdated/stray comment and will just remove it. The check should be relative to  "block_size" as in the check shortly below that:
```py
    if m3 > padded_size // block_size:
        raise ValueError("m3 too large. must be <= arr.shape[1] // 32")
```

I will update that string to use `block_size` instead of 32 in case we change block_size in the future.